src/store/flow/block-types/MobilePrimitives_SelectOneResponseBlockStore.ts

```diff
@@ -34,41 +35,56 @@ const actions: ActionTree<IEmptyState, IRootState> = {
       throw new ValidationException(`Unable to find resource for choice: ${resourceId}`)
     }
 
-    // defaulted to `resourceId` to mitigate empty keys and associated error handling altogether
-    const desiredChoiceKey = snakeCase(get(resource.values[0], 'value')) || resource.uuid
-    const doesChoiceKeyAlreadyExist = desiredChoiceKey in block.config.choices
-    // apply suffix as resourceId when duplicated to prevent overwriting as input is received
-    const suffix = doesChoiceKeyAlreadyExist ? `-${resource.uuid}` : ''
-    Vue.set(block.config.choices, `${desiredChoiceKey}${suffix}`, resource.uuid)
+    // Do not attribute the resource if it has been done already
+    const existingChoiceByResource = find(block.config.choices, {prompt: resource.uuid})
+    if (existingChoiceByResource === undefined) {
+      const allChoices = block.config.choices
+      allChoices.push({
+        prompt: resource.uuid,
+      } as IChoice)
+      Vue.set(block.config, 'choices', allChoices)
```

Comment:
Should we consider using `block_updateConfigByKey` to unify code and simplify debugging (an extra mutation is reflected in devtools)

Reply:
Good point ! Checking if we do even need that, or the push will handle it

Reply:
I've removed it, it was not necessary ;)